src/main/java/hudson/plugins/gradle/injection/GradleBuildScanInjection.java

```diff
@@ -173,6 +107,9 @@ private boolean initScriptChanged(FilePath gradleInitScriptFile) throws IOExcept
     private void cleanup(Node node, String initScriptDirectory) {
         try {
             removeInitScript(node.getChannel(), initScriptDirectory);
+
+            // we still need to clean up all environment variables set in older
+            // versions even though we now set them in EnvironmentContributor
```

Comment:
Maybe just add that this needs to be removed in the future. This is a temporary behavior.